include/glow/Graph/Utils.h

```diff
@@ -132,6 +132,67 @@ class GraphPreOrderVisitor : public PreOrderVisitor {
   }
 };
 
+using Edge = std::pair<Node *, NodeValue>;
```

Comment:
I think original intent was to capture the upward relationship in case of both up and down traversal, and make it easy to access the "destination" node itself by code that uses traverseUp/traverseDown. Also just having input number would complicate down traversal code a bit as we have to figure out from user node which of it's input corresponds to def node.
Vs now we can just do this:

```
Node *childNode = user.getUser();
    if (matchCondition(childNode)) {
      edgeList.push_back(Edge(childNode, currNodeVal));
    }
```

@ksaurabh-cadence  thoughts?

Reply:
IMHO NodeValue is still light weight. As @ayermolo explained this interpretation was simpler to visualize/write traversal in either direction and provided more direct access to the connecting node.